phpBB/assets/javascript/editor.js

```diff
@@ -24,337 +6,382 @@ function helpline(help) {
+			delete attributes.defaultattr;
 		}
 
-		textarea = doc.forms[form_name].elements[text_name];
+		for(attributeName in attributes){
+			attributeStr += attributeName + '="' + attributes[attributeName] + '" ';
+		}
 
-		phpbb.applyCodeEditor(textarea);
-	});
-})(jQuery);
+		if(attributeStr === ' '){
+			attributeStr = '';
+		}
 
+		return '[' + name + attributeStr + ']' + content + 
+				(isSelfClosing ? '[/' + name + ']' : '');
+	},
+
+
+	/**
+	 * @return A javascript object that allows adding parameters and
+	 * @source Based on s9e\TextFormatter\render.js
+	*/
+	xslt: function (xsl){
+		// older IE has its own way of doing it
+		var standardsBrowser = (typeof DOMParser !== 'undefined' && typeof XSLTProcessor !== 'undefined');
+		if (standardsBrowser) {
+			var xslDoc = (new DOMParser).parseFromString(xsl, 'text/xml');
+
+			var processor = new XSLTProcessor();
+			processor.importStylesheet(xslDoc);
+
+			return {
+				'setParameter': function (name, value){
+					if(!value){
+						if(value === ''){
+							value = ' ';
+						}else{
+							value = '';
+						}
+					}
+					processor.setParameter(null, name, value);
+				},
+
+				'transformToFragment' : function (xml, onDocument){
+					var xmlDoc = (new DOMParser).parseFromString(xml, 'text/xml');
+					// NOTE: importNode() is used because of https://code.google.com/p/chromium/issues/detail?id=266305
+					return onDocument.importNode(processor.transformToFragment(xmlDoc, onDocument), true);
+				}
+			};
+		}else{
+			var ieStylesheet = new ActiveXObject('MSXML2.FreeThreadedDOMDocument.6.0');
+			ieStylesheet.async = false;
+			ieStylesheet.validateOnParse = false;
+			ieStylesheet.loadXML(xsl);
+
+			var ieGenerator = new ActiveXObject("MSXML2.XSLTemplate.6.0");
+			ieGenerator.stylesheet = ieStylesheet;
+			var ieTransformer = ieGenerator.createProcessor();
+
+			return {
+				'setParameter' : function (name, value){
+					if(!value){
+						if(value === ''){
+							value = ' ';
+						}else{
+							value = '';
+						}
+					}
+					ieTransformer.addParameter(name, value, '');
+				},
+
+				'transformToFragment' : function (xml, onDocument){
+					var div = onDocument.createElement('div'),
+						fragment = onDocument.createDocumentFragment();
+
+					var ieTargetStylesheet = new ActiveXObject('MSXML2.FreeThreadedDOMDocument.6.0');
+					ieTargetStylesheet.async = false;
+					ieTargetStylesheet.validateOnParse = false;
+					ieTargetStylesheet.loadXML(xml);
+
+					ieTransformer.input = ieTargetStylesheet
+					ieTransformer.transform();
+
+					div.innerHTML = ieTransformer.output;
+					while (div.firstChild){
+						fragment.appendChild(div.removeChild(div.firstChild));
+					}
+
+					return fragment;
+				}
+			};
+		}
+	},
+
+	/**
+	 * Insert HTML in the editor at the selection point
+	 *
+	 */
+	insertHTML: function (editor, start, end){
+		alert("Misconfigured editor. The editor setup code does not override editor.insertHTML");
```

Comment:
I want to use `alert()` because I want it to be as intrusive as possible. I want whomever works on it or the user to get a "slap in the face" that it is wrong as it currently is.
`console.warn` is too hidden, most popin are not intrusive enough.